api/app/zonal_stats.py

```diff
@@ -317,6 +335,11 @@ def intersect_percentage(masked) -> float:
 
     if not geojson_out:
         feature_properties = _extract_features_properties(zones_filepath)
+
+        if filter_by is not None:
+            key, value = filter_by
+            feature_properties = [p for p in feature_properties if str(p[key]) == value]
```

Comment:
probably worth extracting the filtering mechanism and error handling to re-use it here.

Reply:
The filter process is different since in the first one we get the geometries to be passed to the zonal_stats function, and the second one we extract the properties to get from the api